tutoraspects/templates/aspects/apps/superset/pythonpath/openedx_jinja_filters.py

```diff
@@ -44,4 +48,31 @@ def can_view_courses(username, field_name="course_id", **kwargs):
         return NO_COURSES
 
 
+def translate_column(column_name):
+    """
+    Translate a string to the given language.
+    """
+    roles = security_manager.get_user_roles()
+    log.info(f"Roles: {roles}")
+    lang = "en"
+    if roles:
+        for role in roles:
+            role_str = role.name.split(" - ")
+            if len(role_str) > 1:
+                lang = role_str[1]
+                break
+
+    strings = DATASET_STRINGS.get(column_name, [])
+    case_format = """CASE \n {cases} \n ELSE {column_name} \n END"""
+    single_case_format = "WHEN {column_name} = '{string}' THEN '{translation}'"
+    cases = "\n".join(
+        single_case_format.format(column_name=column_name, string=string, translation=get_translation(string, lang))
```

Comment:
When `get_translation` is called, `localization.py` will load all of the translations for all languages into memory. I'm not sure how long that will stick around for, but loading a bunch of files off disk repeatedly might cause performance issues. I doubt we'll have so many strings that it will make much of a dent in server memory, but we might want to compare before and after this change to see.